### PR TITLE
add missing documentation for various crates

### PR DESCRIPTION
**Pull request by tcoratger**
## Description

This pull request addresses missing documentation within various crates across the project. 

It focuses on enhancing code readability and understanding by adding comprehensive header documentation for structs, enums, and their respective fields.

### CODE REVIEW

File: crates/rpc/rpc/src/eth/error.rs [hunk 1]
```diff
@@ -22,44 +22,48 @@ pub type EthResult<T> = Result<T, EthApiError>;
...
     #[error("invalid transaction signature")]
     InvalidTransactionSignature,
+    /// Errors related to the transaction pool
     #[error(transparent)]
     PoolError(RpcPoolError),
+    /// When an unknown block number is encountered
     #[error("unknown block number")]
     UnknownBlockNumber,
     /// Thrown when querying for `finalized` or `safe` block before the merge transition is
-    /// finalized, <https://github.com/ethereum/execution-apis/blob/6d17705a875e52c26826124c2a8a15ed542aeca2/src/schemas/block.yaml#L109>
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/error.rs:39]: I want to keep the reference

File: crates/rpc/rpc/src/eth/error.rs [hunk 2]
```diff
@@ -22,44 +22,48 @@ pub type EthResult<T> = Result<T, EthApiError>;
...
     /// An internal error where prevrandao is not set in the evm's environment
     #[error("prevrandao not in the EVM's environment after merge")]
     PrevrandaoNotSet,
-    /// Excess_blob_gas is not set for Cancun and above.
-    #[error("excess blob gas missing the EVM's environment after Cancun")]
+    /// `excess_blob_gas` is not set for Cancun and above
+    #[error("excess blob gas missing in the EVM's environment after Cancun")]
     ExcessBlobGasNotSet,
-    /// Thrown when a call or transaction request (`eth_call`, `eth_estimateGas`,
-    /// `eth_sendTransaction`) contains conflicting fields (legacy, EIP-1559)
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/error.rs:53]: I want to keep the reference to the rpc calls

**Review by mattsse — CHANGES_REQUESTED** [review]
Amazing, very much appreciated 🙏

only two nits

File: crates/rpc/rpc/src/eth/error.rs (hunk 2 above)
Comment (tcoratger) [reply crates/rpc/rpc/src/eth/error.rs]: done

File: crates/rpc/rpc/src/eth/error.rs (hunk 1 above)
Comment (tcoratger) [reply crates/rpc/rpc/src/eth/error.rs:39]: done

**Review by mattsse — APPROVED** [review]

**Pull request merged**